Admin/Test-AMSI.ps1

```diff
@@ -118,23 +212,52 @@ function Test-AMSI {
         }
         return
     }
+
     if ($CheckAMSIProviders) {
+        CheckWindowsVersionIsW16orOlder
         $AMSI = Get-ChildItem 'HKLM:\SOFTWARE\Microsoft\AMSI\Providers' -Recurse
         $AMSI -match '[0-9A-Fa-f\-]{36}' | Out-Null
         $Matches.Values | ForEach-Object { Get-ChildItem "HKLM:\SOFTWARE\Classes\ClSid\{$_}" | Format-Table -AutoSize }
     }
+
+    $getSO = $null
     if ($EnableAMSI) {
-        Remove-SettingOverride -Identity DisablingAMSIScan -Confirm:$false
-        Get-ExchangeDiagnosticInfo -Process Microsoft.Exchange.Directory.TopologyService -Component VariantConfiguration -Argument Refresh
-        Write-Warning "Remember to restart IIS for this to take affect. You can accomplish this by running .\Test-AMSI.ps1 -RestartIIS"
+        CheckExchangeAndWindowsVersionsSupportedByAMSI
+        $getSO = Get-SettingOverride -Identity DisablingAMSIScan -ErrorAction SilentlyContinue | Where-Object { $_.SectionName -eq 'HttpRequestFiltering' -and $_.Parameters -eq 'Enabled=False' -and $_.Reason -eq 'Disabled via CSS-Exchange Script' }
+        if ( $null -eq $getSO ) {
+            Write-Warning "AMSI is NOT disabled by CSS-Exchange Script on Exchange configuration"
+        } else {
+            Remove-SettingOverride -Identity DisablingAMSIScan -Confirm:$false
+            Get-ExchangeDiagnosticInfo -Process Microsoft.Exchange.Directory.TopologyService -Component VariantConfiguration -Argument Refresh
+            Write-Warning "Remember to restart IIS for this to take affect. You can accomplish this by running .\Test-AMSI.ps1 -RestartIIS"
+        }
         return
     }
+
     if ($DisableAMSI) {
-        New-SettingOverride -Name DisablingAMSIScan -Component Cafe -Section HttpRequestFiltering -Parameters ("Enabled=False") -Reason "Disabled via CSS-Exchange Script"
-        Get-ExchangeDiagnosticInfo -Process Microsoft.Exchange.Directory.TopologyService -Component VariantConfiguration -Argument Refresh
-        Write-Warning "Remember to restart IIS for this to take affect. You can accomplish this by running .\Test-AMSI.ps1 -RestartIIS"
+        CheckExchangeAndWindowsVersionsSupportedByAMSI
+        $getSO = Get-SettingOverride -Identity DisablingAMSIScan -ErrorAction SilentlyContinue | Where-Object { $_.SectionName -eq 'HttpRequestFiltering' -and $_.Parameters -eq 'Enabled=False' -and $_.Reason -eq 'Disabled via CSS-Exchange Script' }
```

Comment:
Same here. We don't really care about the Reason. The question is whether it is set or not.